auth/dynamo_token_manager.go

```diff
@@ -20,14 +22,92 @@ func NewDynamoTokenManager(table string, dynamodb dynamodbiface.DynamoDBAPI) *Dy
 }
 
 func (d *DynamoTokenManager) CreateToken(user string) (string, error) {
-	return "", fmt.Errorf("CreateToken not implemented")
+	token := convertToToken(randomString(20), randomString(20))
+
+	item := map[string]*dynamodb.AttributeValue{
+		"User": {
+			S: &user,
+		},
+		"Token": {
+			S: &token,
+		},
+	}
+
+	input := &dynamodb.PutItemInput{}
+	input.SetTableName(d.table)
+	input.SetItem(item)
+
+	if err := input.Validate(); err != nil {
+		return "", err
+	}
+
+	if _, err := d.dynamodb.PutItem(input); err != nil {
+		return "", err
+	}
+
+	return token, nil
 }
 
 func (d *DynamoTokenManager) DeleteToken(token string) error {
-	return fmt.Errorf("DeleteToken not implemented")
+	key := map[string]*dynamodb.AttributeValue{
+		"Token": {
+			S: &token,
+		},
+	}
+
+	input := &dynamodb.DeleteItemInput{}
+	input.SetTableName(d.table)
+	input.SetKey(key)
+
+	if err := input.Validate(); err != nil {
+		return err
+	}
+
+	if _, err := d.dynamodb.DeleteItem(input); err != nil {
+		return err
+	}
+
+	return nil
 }
 
 func (d *DynamoTokenManager) Authenticate(user, pass string) (bool, error) {
-	log.Println("[ERROR] - DynamoTokenManager.Authenticate not implemented")
-	return true, nil
+	token := convertToToken(user, pass)
+
+	key := map[string]*dynamodb.AttributeValue{
+		"Token": {
+			S: &token,
+		},
+	}
+
+	input := &dynamodb.GetItemInput{}
+	input.SetTableName(d.table)
+	input.SetConsistentRead(true)
+	input.SetKey(key)
+
+	if err := input.Validate(); err != nil {
+		return false, err
+	}
+
+	output, err := d.dynamodb.GetItem(input)
```

Comment:
> That token should be good for immediate use and valid for authentication, right?
Yes, if they put the token into their `~/.docker/config.json` file (or `dockercfg`)

> Wouldn't that token fail here in Authenticate? It seems like we're looking for a "user:pass" token when the token stored in Create is "rando:rando".
Nope - it would pass for the reason I stated above. When you put it into the `config.json` file, the `docker` cli passes the token along as basic auth, e.g. they pass along `rando:rando` when they use the token.